File: src/main/java/SphereEngine/Api/compilers/submissions/getSubmission.java
package SphereEngine.Api.compilers.submissions;
/**
 * Example presents usage of the successful getSubmission() API method  
 */

import SphereEngine.Api.CompilersClientV3;
import SphereEngine.Api.Exception.NotAuthorizedException;
import SphereEngine.Api.Exception.NotFoundException;
import SphereEngine.Api.Exception.ClientException;
import SphereEngine.Api.Exception.ConnectionException;
import com.google.gson.JsonObject;

public class getSubmission
{

	public static void main(String[] args) 
	{
		CompilersClientV3 client = new CompilersClientV3(
				"8bce219ca0c54390d6b832fa542470b2", 
				"a53fdb4d.compilers.sphere-engine.com");
		
		try {
			JsonObject response = client.getSubmission(82647342);
			System.out.println(response);
		} catch (NotAuthorizedException e) {
			System.out.println("Invalid access token");
		} catch (NotFoundException e) {
			System.out.println("Submission does not exist");
		} catch (ClientException e) {
			System.out.println(e.getMessage());
		} catch (ConnectionException e) {
			System.out.println(e.getMessage());
		}
	}	
}

File: src/main/java/SphereEngine/get_submission.py
#!/usr/bin/env python2
#encoding: UTF-8

"""
Example presents error handling for submissions.get() API method
"""
from sphere_engine import CompilersClientV4
from sphere_engine.exceptions import SphereEngineException

# define access parameters
accessToken = '8bce219ca0c54390d6b832fa542470b2'
endpoint = 'a53fdb4d.compilers.sphere-engine.com'

# initialization
client = CompilersClientV4(accessToken, endpoint)

# API usage
try:
    response = client.submissions.get(82807704)
    print(response)
except SphereEngineException as e:
    if e.code == 401:
        print('Invalid access token')
    elif e.code == 403:
        print('Access to the submission is forbidden')
    elif e.code == 404:
        print('Submission does not exist')
